hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/converter/FSConfigToCSConfigConverter.java

```diff
@@ -412,6 +418,20 @@ private void emitDefaultMaxAMShare() {
           queueMaxAMShareDefault);
     }
   }
+
+  private void emitDefaultUserLimitFactor(Collection<FSLeafQueue> fsLeafQueue) {
+    fsLeafQueue
+        .forEach((leafQueue) -> {
+          if (!capacitySchedulerConfig.
+                isAutoQueueCreationV2Enabled(leafQueue.getName())) {
+            capacitySchedulerConfig.setFloat(
+                    CapacitySchedulerConfiguration.
+                            PREFIX + leafQueue.getName() + DOT + USER_LIMIT_FACTOR,
+                    -1.0f);
+          }
+        });
+  }
+
```

Comment:
On the second look this class is more like for general config changes, while there is a class called FSQueueConverter, which has the queue specific converter logic implemented. Can you please move this whole logic there?